demos/noise_suppression_demo/cpp/main.cpp

```diff
@@ -0,0 +1,336 @@
+    short data_format;           /* Data format tag, 1 = PCM */
+    short num_of_channels;       /* Number of channels in file */
+    int sampling_freq;           /* Sampling frequency */
+    int bytes_per_sec;           /* Average bytes/sec */
+    short block_align;           /* Block align */
+    short bits_per_sample;
+    unsigned int data_tag;       /* "data" string */
+    int data_length;             /* Raw data length */
+} RiffWaveHeader;
+
+
+bool read_wav(const std::string& file_name, std::vector<short>* p_wave, RiffWaveHeader* p_wave_header)
+{
+    FILE *inp_wave = fopen(file_name.c_str(), "rb");
+    if (inp_wave == NULL)
+    {
+        std::cerr << "fail to read " << file_name << std::endl;
+        return false;
+    }
+
+    if (1 != fread(p_wave_header, sizeof(RiffWaveHeader), 1, inp_wave))
+    {
+        std::cerr << "fail to read header for " << file_name << std::endl;
+        return false;
+    }
+
+    bool read_ok = true;
+    // make sure it is actually a RIFF file
+    if (0 != memcmp(&p_wave_header->riff_tag, "RIFF", 4))
+    {
+        std::cerr << "riff_tag != 'RIFF' for " << file_name << std::endl;
+        read_ok = false;
+    }
+    if (0 != memcmp(&p_wave_header->wave_tag, "WAVE", 4))
+    {
+        std::cerr << "wave_tag != 'WAVE' for " << file_name << std::endl;
+        read_ok = false;
+    }
+    if (0 != memcmp(&p_wave_header->fmt_tag, "fmt ", 4))
+    {
+        std::cerr << "fmt_tag != 'fmt' for " << file_name << std::endl;
+        read_ok = false;
+    }
+
+    // only PCM
+    if (p_wave_header->data_format != 1)
+    {
+        std::cerr << "data_format != kPCMFormat(1) for " << file_name << std::endl;
+        read_ok = false;
+    }
+    // only mono
+    if (p_wave_header->num_of_channels != 1)
+    {
+        std::cerr << "num_of_channels != 1 for " << file_name << std::endl;
+        read_ok = false;
+    }
+    // only 16 bit
+    if (p_wave_header->bits_per_sample != 16)
+    {
+        std::cerr << "bits_per_sample != 16 for " << file_name << std::endl;
+        read_ok = false;
+    }
+    // only 16KHz
+    if (p_wave_header->sampling_freq != 16000)
+    {
+        std::cerr << "sampling_freq != 16000 for " << file_name << std::endl;
+        read_ok = false;
+    }
+    // make sure that data chunk follows file header
+    if (0 != memcmp(&p_wave_header->data_tag, "data", 4))
+    {
+        std::cerr << "data_tag != 'data' for " << file_name << std::endl;
+        read_ok = false;
+    }
+
+    if (!read_ok)
+    {
+        return false;
+    }
+
+    size_t wave_size = p_wave_header->data_length / 2;
+    p_wave->resize(wave_size);
+
+    if (1 != fread(&(p_wave->front()), wave_size*2, 1, inp_wave))
+    {
+        std::cerr << "fail to read data for " << file_name << std::endl;
+        return false;
+    }
+
+    return true;
+}
+
+bool write_wav(const std::string& file_name, const std::vector<short>& wave, const RiffWaveHeader& wave_header)
+{
```

Comment:
```suggestion
bool write_wav(const std::string& file_name, const std::vector<short>& wave, const RiffWaveHeader& wave_header) {
```
We place brackets on the same line